README.md

```diff
@@ -6,66 +6,123 @@ Please, read following instructions and let us know if anything is not clear.
+- [ ] **F6**: Remove a todo item
+  - After clicking the `ListItemRemoveButton` component in the `ListItem`, the todo item should be removed.
+  - Changes to the data should be persisted on the server via API calls.
+  - The list of the todo items on the client should be updated.
+- [ ] **F7**: Sort the todo items
+  - Sort the list of the todo items:
+    - "todo" items (not "done") should be displayed first,
+    - after that, items should be sorted by their creation date, descending.
+- [ ] **F8**: Count the todo items
+  - Show a number of the "todo"/"done" items in the `Footer`. 
+- [ ] **F9**: `Button` component
+  - Create a `Button` component and use it instead of HTML `button` elements.
+
+#### Styling
+
+- [ ] **UI1**: `HeaderAddButton` alignment
+  - `HeaderAddButton` component should be right-aligned in the `Header` component.
+- [ ] **UI2**: `ListItem` actions alignment
+  - Action buttons in the `ListItem` component should be right-aligned.
+- [ ] **UI3**: `ListItem` actions visibility
+  -  Action buttons in the `ListItem` component should be visible only when hovering over the `ListItem`.
+
+#### Stories
+
+- [ ] **SB1**: Add a story/stories for the `Layout` component.
+- [ ] **SB2**: Add a story showing the `ListItem` with visible buttons (implemented in _"UI3"_).
+- [ ] **SB3**: Add stories showing available `Button` variations.
 
 ### Server
-- [ ] **S1**: Implement custom endpoint for marking single todo item as "done". Calling this endpoint sets `done` field to `true` and `finishedAt` field sets to current time. Use this new endpoint on client
+
+- [ ] **S1**: Custom "done" endpoint 
+  - Implement a custom API endpoint for setting/un-setting a single todo item as "done".
+  - Calling this endpoint sets the `isDone` field to `true`/`false`.
+  - Additionally, a new field `finishedAt` should be populated. Value should be the current time when the item is "done" and `null` when it is not "done".
+  - Use this new endpoint in the client.
 
 ### Advanced tasks (optional)
-- [ ] **O1**: modify the `Form` component (and related ones if needed) so the form field handles not only string, but also number and treat empty string input as `undefiened`
-- [ ] **O2**: limit amount of server calls needed to necessary minimum
 
-## Additional comments and restrictions:
-- do not modify API (props) of provided components unless achieving required tasks would not be possible otherwise,
-- do not use any component library like Material UI; do not use Tailwind,
-- feel free to do any other visual modifications that - in your opinion - makes it looking nicer,
-- feel free to install and use any public package you may need
+- [ ] **O1**: Modify the `Form` component (and other related components if needed) so the form field handles not only string values, but also numbers. Treat empty string input as `undefined`.
+- [ ] **O2**: Limit the amount of server calls needed to a necessary minimum.
```

Comment:
I am not sure how would I achieve this? Isn't this more of a requirement (or suggestion) than a task?

Reply:
If you mean O2, it is really more non-functional requirement. I think the idea was eg. not to have server call on every component re-render. But I think we can remove it from here. It typically wasn't what people had in their assignments.